src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+public class Parser {
+    
+    /**
+     * Returns a Command with enum type by parsing the string of command that the user inputs.
+     *
+     * @param input the string represents command that the user inputs.
+     * @return a Command of the enum type.
+     */
+    public static Command parseCommand(String input) {
+        return Command.valueOf(input.toUpperCase(Locale.ROOT));
+    }
+
+    /**
+     * Returns a task by parsing description under instruction of command.
+     *
+     * @param command the Command enum type.
+     * @param input input of user.
+     * @return the task corresponding to the input.
+     * @throws ParseException if parsing fails.
+     */
+    public static Task parseDescription(Command command, String input) throws ParseException {
+        switch (command) {
+        case TODO:
+            return parseToDo(input);
+        case DEADLINE:
+            return parseDeadline(input);
+        case EVENT:
+            return parseEvent(input);
+        default:
+            throw new ParseException("the first argument command is not valid.");
+        }
+    }
+
+    /**
+     * Returns a ToDo by parsing the input behind the command.
+     * Used when the command is todo.
+     * Works when the input is not empty.
+     *
+     * @param input input of users.
+     * @return a task of ToDo corresponding to the input.
+     * @throws ParseException if the description is empty.
+     */
+    private static ToDo parseToDo(String input) throws ParseException {
+        if (input.isEmpty() || input.equals(" ")) {
+            throw new ParseException("OOPS!!! The description of a todo cannot be empty.\n");
+        }
+        if (input.charAt(0) == ' ') {
+            input = input.substring(1);
+        }
+        return new ToDo(input);
+    }
+
+    /**
+     * Returns a Deadline by parsing the input behind the command.
+     * Used when the command is deadline.
+     * Works when the input is "{description} /by YYYY-MM-DD".
+     *
+     * @param input input of users.
+     * @return a task of Deadline corresponding to the input.
+     * @throws ParseException if the description is empty.
+     * @throws DateTimeParseException if the format of data time is not correct.
+     */
+    private static Deadline parseDeadline(String input) throws ParseException, DateTimeParseException {
```

Comment:
I see the `parseDeadline` and `parseEvent` has quite a number of similarities, do you think can these 2 be placed under one method instead? 🤔

Reply:
Sure. I think it will be more consice to place them in one method. Thank you!